### PR TITLE
[consensus] let consensus call LEC

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

According to #3744 , consensus has to talk to LEC directly instead of building a block_executor.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

CI

### CODE REVIEW

File: consensus/Cargo.toml [hunk 1]
```diff
@@ -46,6 +47,7 @@ libra-types = { path = "../types", version = "0.1.0" }
 libra-workspace-hack = { path = "../common/workspace-hack", version = "0.1.0" }
 network = { path = "../network", version = "0.1.0" }
 safety-rules = { path = "safety-rules", version = "0.1.0" }
+simple-storage-client = { path = "../storage/simple-storage-client", version = "0.1.0" }
```
Comment (zekun000) [inline consensus/Cargo.toml:50]: where do you use this?

File: consensus/src/state_computer.rs [hunk 2]
```diff
@@ -21,14 +22,17 @@ use std::{
 /// Basic communication with the Execution module;
 /// implements StateComputer traits.
 pub struct ExecutionProxy {
-    executor: Mutex<Box<dyn BlockExecutor>>,
+    lec: Mutex<Box<dyn BlockExecutor + Send + Sync>>,
```
Comment (zekun000) [inline consensus/src/state_computer.rs:25]: this is a worse name

File: consensus/src/consensus_provider.rs [hunk 3]
```diff
@@ -42,7 +41,11 @@ pub fn start_consensus(
         .expect("Failed to create Tokio runtime!");
     let storage = Arc::new(StorageWriteProxy::new(node_config, libra_db));
     let txn_manager = Box::new(MempoolProxy::new(consensus_to_mempool_sender));
-    let state_computer = Arc::new(ExecutionProxy::new(executor, state_sync_client));
+    let execution_correctness_manager = ExecutionCorrectnessManager::new(node_config);
```
Comment (davidiw) [inline consensus/src/consensus_provider.rs:44]: the copypasta is strong with this one.

File: config/src/config/execution_config.rs [hunk 4]
```diff
@@ -45,7 +45,7 @@ impl Default for ExecutionConfig {
         ExecutionConfig {
             genesis: None,
             genesis_file_location: PathBuf::new(),
-            service: ExecutionCorrectnessService::Local,
+            service: ExecutionCorrectnessService::Thread,
```
Comment (davidiw) [inline config/src/config/execution_config.rs:48]: I will admit I'm a little reserved about this one.... I think flipping it now to evaluate is okay, but we should revisit where we are in a few weeks.

Comment (lightmark) [reply config/src/config/execution_config.rs:48]: What's wrong with this than local? At least this one includes ser/de and I think `safety_rules` uses it too.

File: consensus/Cargo.toml (hunk 1 above)
Comment (lightmark) [reply consensus/Cargo.toml:50]: I think I forget to remove it

**Pull request closed without merging**